index.js

```diff
@@ -213,21 +216,25 @@ function isFastifyMultipartRegistered (fastify) {
   return fastify.hasContentTypeParser('multipart') && fastify.hasRequestDecorator('multipart')
 }
 
-function createRequestRetry (requestImpl, reply, retriesCount, retryOnError) {
+function createRequestRetry (requestImpl, reply, retriesCount, retryOnError, retryOnCode = 503) {
   function requestRetry (req, cb) {
     let retries = 0
 
     function run () {
       requestImpl(req, function (err, res) {
-        if (err && !reply.sent && retriesCount > retries) {
-          if (err.code === retryOnError) {
+        // Magic number, so why not 42? We might want to make this configurable.
+        let retryAfter = 42
```

Comment:
```suggestion
        let retryAfter = 42 * Math.random() * (retries + 1)
```

This is probably better as it is not fixed.